docs/actions.md

```diff
@@ -513,6 +514,35 @@ $ wsk action invoke --blocking --result helloPython --param name World
   }
 ```
 
+### Packaging Python actions in zip files
+
+You can package a Python action and dependant modules in a zip file.
+
+The virtual environment package *virtualenv* is supported to pass pip installed packages together with the action in the zip file.
+Package installations inside virtualenv must be done in the target environment (alpine, python 2.7), so a docker image (e.g. jfloff/alpine-python:2.7) is used to create the virtualenv directory. 
+
+Naming conventions:
+
+1. The name of the main Python file must be `__main__.py`
+2. The directory name for a virtualenv directory must be `virtualenv`
+
+Creating the zip file:
+
+1. name your main Python file `__main__.py`
+2. Change into docker image: `docker run -it -v "$PWD:/tmp" jfloff/alpine-python:2.7 bash`
+3. change into the shared folder: `cd tmp`
+4. install virtualenv: `pip install virtualenv`
+5. create the virtual environment named 'virtualenv': `virtualenv virtualenv`
+6. activate the virtual environment: `source virtualenv/bin/activate`
+7. install all required modules `e.g. pip install xxx`
+8. deactivate the virtual environment: `deactivate`
+9. exit docker image: `exit`
```

Comment:
Why don't we do all these steps in one single command? Advocate the use of `requirements.txt` and then do:

```
docker run -v "$PWD:/tmp" python:2.7.13-alpine sh -c "pip install virtualenv && cd /tmp && virtualenv virtualenv && source virtualenv/bin/activate && pip install -r requirements.txt"
```

Should be perfectly fine and far less manual steps.